lua/tardis/sh_parts.lua

```diff
@@ -53,7 +53,19 @@ local overrides={
         local int=self.interior
         local ext=self.exterior
         if self._init and IsValid(int) and IsValid(ext) then
-            if (int:CallHook("ShouldThink")~=false) or (ext:DoorOpen() and self.ClientThinkOverride and LocalPlayer():GetPos():Distance(ext:GetPos())<TARDIS:GetSetting("portals-closedist")) or self.ExteriorPart then -- TODO: Improve
+            local think_ok = (int:CallHook("ShouldThink") ~= false)
+
+            local function is_visible_through_door()
+                local door = ext:DoorOpen()
```

Comment:
This is much better for readability, but I am concerned all the local variable assignments may affect performance - would you mind checking to see if there is any noticeable drop in framerate for an interior with lots of parts from the original version? 

If so, we could still split it out onto multiple lines but maybe use comments instead to denote what's going on

Reply:
The performance isn't affected much for me, but I've made the condition a bit better

Reply:
588f3e54ca30455fb4b173cca793071033e08ce4 should make it optimal